src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -482,8 +502,11 @@ public object CreateFileResult(IFileStoreEntry mediaFile)
             };
         }
 
-        public IActionResult MediaApplication(MediaApplicationViewModel model)
+        public async Task<IActionResult> MediaApplication(MediaApplicationViewModel model)
         {
+            // Check if the user has access to new folders. If not, we hide the "create folder" button from the root folder.
+            model.AllowNewRootFolders = !HttpContext.IsSecureMediaEnabled() || await _authorizationService.AuthorizeAsync(User, SecureMediaPermissions.ViewMedia, (object)"_non-existent-path-87FD1922-8F88-4A33-9766-DA03E6E6F7BA");
```

Comment:
I just noticed the GUID, what's used for? can we make it constant?

Reply:
It's just to make sure no one will ever accidentally creates a real folder with that name. It's never used anywhere for another purpose, no need to make it a global constant.

Reply:
Got it but why this GUID specifically?!!

Reply:
It's just generated on my machine.

Reply:
@Piedone is this GUID for you, coz it doesn't make sense to me, or we might need to generate a new one instead of reling on static one

Reply:
What this does is just disable the UI button to create new folders, if a new folder is not accessible for the current user after creating it. As said, it only disables the button, it is not security critical because the permissions are checked anyway. 

What's inside this string doesn't actually matter at all, but it should not be something users could easily guess. To add some randomness I just appended a GUID, but you should not think of it as a normal GUID...

Reply:
And it is fine so. This GUID is not "for me" :).